README.md

```diff
@@ -28,22 +151,151 @@ mtx.myTranspose();
        ]
 ```
 
+Array.prototype.myTranspose = function(callback){
+  let grid = [];
+  for(let i = 0; i < mtx[0].length; i++){
+    grid[i] = []
+
+    for(let j = 0; j< mtx.length;j++){
+      grid[i][j] = mtx[j][i]
+
+    }
+  }
+  return grid
+};
+
+let mtx = [
+            [1, 2],
+            [3, 4],
+            [5, 6]
+          ]
+
+//console.log(transpose(mtx))
+let results = mtx.myTranspose((el) => {
+  console.log(el)
+})
+console.log(results)
+
 7. Use `myMap` to build a new array where every element is incremented by 10.
-8. Use `myMap` to build a new array where every non-string element is converted to an empty string. 
+
+Array.prototype.myForEach = function (callback) {
+  for(let i = 0; i < this.length; i++){
+    callback(this[i]);
+  }
+
+};
+
+Array.prototype.myMap = function(callback) {
+let newArray = [];
+this.myForEach((el) => {
+  newArray.push(callback(el))
+})
+return newArray
+};
+
+
+/*let arr2 = [1,2,3,4,5]
+let finalArray = arr2.myMap((el) => {
+  return el * 2
+
+});*/
+
+let arr = [1,2,3,4,5,6];
+
+let results = arr.myMap((el) => {
+  console.log(el * 10)
+});
+
+
+console.log(results)
+
+8. Use `myMap` to build a new array where every non-string element is converted to an empty string.
+Array.prototype.myForEach = function (callback) {
+  for(let i = 0; i < this.length; i++){
+    callback(this[i]);
+  }
+
+};
+
+Array.prototype.myMap = function(callback) {
+let newArray = [];
+this.myForEach((el) => {
+  newArray.push(callback(el))
+})
+return newArray
+};
+
+let arr = ["hello","world",1,2,"how",4]
+
+let results = arr.myMap((el)=> {
+if(typeof el === "number"){
+  return ""
+}else{
+  return el
+}
+return results
+})
+
+
+console.log(results)
 9. Use `myFilter` to get only the even elements in an array.
-10. Use `myEvery` to check if all elements in the array are the same. 
-11. Use `myReduce` to return the sum of every element in an array. 
-12. Use `myMap` to build a new array that doubles each element. Then chain `myReduce` to find the product of all the elements. 
-13. Use `myReduce` to find the largest number in an array. 
+
+Array.prototype.myForEach = function (callback) {
+  for(let i = 0; i < this.length; i++){
+    callback(this[i]);
+  }
+
+}
+
+
+Array.prototype.myMap = function(callback) {
+let newArray = [];
+this.myForEach((el) => {
+  newArray.push(callback(el))
+})
+return newArray
+};
+
+Array.prototype.myFilter = function(callback){
+ let filArr = [];
+ this.myForEach(el => {
+  if(callback){
+   filArr.push(callback(el));
+  }
+ });
+ return filArr;
+}
+  let arr = [1,2,3,4,5,6,7,8,9]
+
+  let results = arr.myFilter((el) => {
+    if(el % 2 === 0){
+      return el
+    }
+  });
+
+console.log(results);
+
+
+10. Use `myEvery` to check if all elements in the array are the same.
+let arr = [1,2,3,4,5];
+let results = arr.myEvery((el) => {
+  if(el % 2 === 0){
+  return true
+}
+});
+console.log(results)
+11. Use `myReduce` to return the sum of every element in an array.
```

Comment:
It's really useful for us to have something for these questions, even if it's just an attempt. Help us help you, OK?